contracts/oracles/IporOracle.sol

```diff
@@ -165,16 +158,25 @@ contract IporOracle is
         uint256 length = indexesToUpdate.length;
         require(length > 0, IporErrors.INPUT_ARRAYS_LENGTH_MISMATCH);
         for (uint256 i; i < length; ) {
-            if (indexesToUpdate[i].asset == _stEth) {
-                _updateIndexAndQuasiIbtPrice(
-                    indexesToUpdate[i].asset,
-                    indexesToUpdate[i].indexValue,
-                    indexesToUpdate[i].updateTimestamp,
-                    indexesToUpdate[i].quasiIbtPrice
-                );
-            } else {
-                _updateIndex(indexesToUpdate[i].asset, indexesToUpdate[i].indexValue, block.timestamp);
+            _updateIndex(indexesToUpdate[i].asset, indexesToUpdate[i].indexValue, block.timestamp);
+            unchecked {
+                ++i;
             }
+        }
+    }
+
+    function updateIndexesAndQuasiIbtPrice(
+        IIporOracle.UpdateIndexParams[] calldata indexesToUpdate
+    ) external override onlyUpdater whenNotPaused {
+        uint256 length = indexesToUpdate.length;
+        require(length > 0, IporErrors.INPUT_ARRAYS_LENGTH_MISMATCH);
```

Comment:
This validation is redundant - it had sense when we have two arrays in input param, now we have only one array with structure.